elastic/kibana
# [Synthetics] enable/disable - prevent incorrect keys from being added to the monitor saved object

## Summary

Fixes a bug where incorrect keys were being added to montiors on the enable/disable flow in Synthetics UI.

This was due to an oversight where types were changed to accept either a Synthetics Monitor saved object or an Overview Monitor Item. In reality, Overview Monitor Items are not a type of Synthetics monitor, but is a unique object for displaying information about a monitor on the overview page in general. This only impacted 8.5.0.

The enable/disable hook was updated to adjust the types.

Additionally, [extra runtime type decoding](https://github.com/elastic/kibana/pull/140553/files#diff-565557ca037738094f191d1c22392e7e845077c193b2e89079152930eedb6792R76-R94) was added to ensure that unknown keys cannot be added to a monitor-saved object. The `t.exact` type returns a valid monitor, stripped of any additional unknown keys that are not part of the codec.


### Testing
1. Create a monitor.
2. Navigate to Overview. Disable the monitor
3. Inspect the monitor saved object. You can do this by viewing the monitor get request when editing the monitor. 
4. Inspect the `attributes` key of the monitor. Ensure that no keys from the [`MonitorOverviewItem`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/common/runtime_types/monitor_management/monitor_types.ts#L383) are present, such as `location`, `isEnabled` or `id`.
